src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,204 @@
+"""
+File       : MSPileupData.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupData class provides whole logic behind
+data used and stored by MSPileup module.
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# system modules
+import time
+import calendar
+
+# third party modules
+from pymongo import IndexModel, errors
+
+# WMCore modules
+from WMCore.Database.MongoDB import MongoDB
+from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.MSPileup.DataStructs.MSPileup import MSPileupObj, gmtime
+from WMCore.MicroService.MSPileup.MSPileupError import MSPileupError
+
+
+def stripKeys(docs, skeys=None):
+    """
+    Helper function to strip out keys from given dictionary. The input parameter
+    is a dictionary and keys are wiped out in place
+    :param docs: either input dictionary or list of dictionaries
+    :param skeys: list of of strip keys
+    """
+    if skeys is None:
+        return docs
+    if isinstance(docs, list):
+        results = []
+        for doc in docs:
+            if doc:
+                for key in skeys:
+                    doc.pop(key, None)
+                    results.append(doc)
+        return results
+
+    if docs and isinstance(docs, dict):
+        for key in skeys:
+            docs.pop(key, None)
+    return docs
+
+
+def encodeTimestamps(doc):
+    """
+    Encode timestamps in MSPileup data document
+    """
+    for key in ["insertTime", "lastUpdateTime"]:
+        val = doc.get(key, None)
+        if val and isinstance(val, int):
+            # convert to string representation
+            doc[key] = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(val))
+    return doc
+
+
+def decodeTimestamps(doc):
+    """
+    Decode timestamps in MSPileup data document
+    """
+    for key in ["insertTime", "lastUpdateTime"]:
+        val = doc.get(key, None)
+        if val and isinstance(val, str):
+            # convert to seconds since epoch in GMT time zone
+            doc[key] = calendar.timegm(time.strptime(val, "%Y-%m-%dT%H:%M:%SZ"))
+    return doc
+
+
+class MSPileupData(MSCore):
```

Comment:
For this class, I would highly recommend to use more meaningful (and less confusing) method names, e.g.:
* get --> getPileup
* create --> createPileup
* and so on.

Reply:
Second commit. I see not reason why it should inherit from MSCore. This is only dealing with MongoDB, nothing else and it's not going to be the MSPileup daemon/app, AFAIU.

Reply:
done